### PR TITLE
More clang tidy

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

File: hl/examples/ex_table_04.c [hunk 1]
```diff
@@ -65,23 +65,23 @@ main(void)
     hid_t     string_type;
     hid_t     file_id;
     hsize_t   chunk_size   = 10;
-    Particle  fill_data[1] = {{"no data", -1, -1, -99.0f, -99.0}}; /* Fill value particle */
+    Particle  fill_data[1] = {{"no data", -1, -1, -99.0F, -99.0}}; /* Fill value particle */
```
Comment (byrnHDF) [inline hl/examples/ex_table_04.c:68]: Didn't we just change all these to lower case "f" earlier this year?

Comment (seanm) [reply hl/examples/ex_table_04.c:68]: I don't know. I don't follow HDF5 closely. The clang-tidy docs are [here](https://clang.llvm.org/extra/clang-tidy/checks/readability-uppercase-literal-suffix.html) but they don't really give a rationale.  I think it's mostly for `l` which can look a lot like `1`. I guess the idea is to consistently use uppercase. I can drop this commit, I don't have strong feelings about it for `f` vs `F`, but for `l` vs `L` I do think uppercase is better.

Comment (byrnHDF) [reply hl/examples/ex_table_04.c:68]: I am not entirely sure myself, but I think it was a gcc warning issue.

Comment (seanm) [reply hl/examples/ex_table_04.c:68]: I've just checked the `develop` branch, and there is a mix of `f` and `F` currently.

Comment (byrnHDF) [reply hl/examples/ex_table_04.c:68]: Unsurprising.

Comment (seanm) [reply hl/examples/ex_table_04.c:68]: So using `readability-uppercase-literal-suffix` as per this patch would have the advantage of making it uniform, even though `F` vs `f` is not as useful as `l` vs `L`.

Comment (byrnHDF) [reply hl/examples/ex_table_04.c:68]: We did a quick check and it seems we have no standard for F vs f.  We will need to decide so we don't keep churning.

**Review by byrnHDF — APPROVED** [review]

**Review by lrknox — APPROVED** [review]

File: hl/examples/ex_table_04.c (hunk 1 above)
Comment (derobins) [reply hl/examples/ex_table_04.c:68]: We have no standard, but I'm fine with upper case F being the standard, especially given the l vs L confusion.

**Review by derobins — APPROVED** [review]

**Pull request merged**